File: 2022/C++/Day-1/Part-2.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream inputFile("input.txt");
    std::ios_base::sync_with_stdio(false);
    //-------------------------------------
    int first(0), second(0), third(0);
    
    std::string line;
    int sum(0);
    while (std::getline(inputFile, line)) {
        if (!line.empty()) {
            sum += std::stoi(line);
        } else {
            if (sum > first) {
                third = second;
                second = first;
                first = sum;
            } else if (sum > second) {
                third = second;
                second = sum;
            } else if (sum > third) {
                third = sum;
            }

            sum = 0;
        }
    }
    first = std::max(first, sum);
    

    std::cout << first + second + third << "\n";
    //-------------------------------------
    return 0;
}

File: 2022/Python/Day-1/Part-2.py
from helper_functions import get_relative_path


def main():
    with open(get_relative_path("input.txt")) as file:
        lines = file.read().split("\n\n")
        sums = [sum(map(int, line.split("\n"))) for line in lines]
        sums.sort()
    print(sum(sums[-3:]))


if __name__ == "__main__":
    main()
